In [1]:
import pandas as pd
import numpy as np
import re

In [2]:
df = pd.read_csv("AC_Data_Final.csv")
df

Unnamed: 0   Brands                                              Model  \
0             0    ONIDA  ONIDA 2023 Model 1.5 Ton 3 Star Split Inverter...   
1             1   Godrej  Godrej 5-in-1 Convertible Cooling 2023 Model 1...   
2             2   Voltas  Voltas 2023 Model 0.6 Ton 2 Star Split AC  - W...   
3             3   Voltas  Voltas 2023 Model 1.5 Ton 5 Star Split Inverte...   
4             4   Godrej  Godrej 5-In-1 Convertible Cooling 2023 Model 1...   
..          ...      ...                                                ...   
750         750       LG  LG 1.5 Ton 5 Star Split Dual Inverter AC  - White   
751         751   Daikin  Daikin 1.5 Ton 3 Star Split with PM 2.5 Filter...   
752         752  Hitachi    Hitachi 1 Ton 3 Star Split Inverter AC  - White   
753         753     Blue  Blue Star Convertible 5 in 1 Cooling 2023 Mode...   
754         754     Blue  Blue Star 2023 Model 1.3 Ton 3 Star Split Inve...   

     BEE_Rating  Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
0           3.0   1.5           939         7,988          4.3        ₹43,900   
1           4.0   1.5           403         4,378          4.1        ₹49,990   
2           2.0   0.6            13           133          4.0        ₹49,690   
3           5.0   1.5           244         2,635          4.1        ₹75,990   
4           3.0   1.5           403         4,378          4.1        ₹45,400   
..          ...   ...           ...           ...          ...            ...   
750         5.0   1.5             1            18          4.2        ₹83,990   
751         3.0   1.5           204         2,045          4.3        ₹43,800   
752         3.0   1.0             4            23          3.6            NaN   
753         3.0   1.5           384         4,478          4.1        ₹62,250   
754         3.0   1.3         1,521        13,889          4.2        ₹57,500   

    Current_Price  Room_Size  Annual_Power_Usage  
0         ₹29,990  111 - 150              969.15  
1         ₹32,990  111 - 150              883.14  
2         ₹19,999         90              475.30  
3         ₹37,990  111 - 150              751.28  
4         ₹30,990  111 - 150              965.71  
..            ...        ...                 ...  
750       ₹51,890  111 - 150              968.61  
751       ₹40,288  111 - 150             1425.00  
752       ₹41,290         90                0.00  
753       ₹35,799  111 - 150             1002.84  
754       ₹32,990  111 - 150              865.48  

[755 rows x 12 columns]

In [3]:
df.columns

Index(['Unnamed: 0', 'Brands', 'Model', 'BEE_Rating', 'Tons', 'Reviews_Count',
       'Ratings_Count', 'Cust_Review', 'Original_Price', 'Current_Price',
       'Room_Size', 'Annual_Power_Usage'],
      dtype='object')

In [4]:
df.isnull().sum()

Unnamed: 0              0
Brands                  0
Model                   0
BEE_Rating              3
Tons                    1
Reviews_Count           0
Ratings_Count           0
Cust_Review             0
Original_Price        100
Current_Price           0
Room_Size               4
Annual_Power_Usage     19
dtype: int64

### Removing the old index column

In [5]:
df.drop(columns="Unnamed: 0",inplace=True)

In [6]:
df

Brands                                              Model  BEE_Rating  \
0      ONIDA  ONIDA 2023 Model 1.5 Ton 3 Star Split Inverter...         3.0   
1     Godrej  Godrej 5-in-1 Convertible Cooling 2023 Model 1...         4.0   
2     Voltas  Voltas 2023 Model 0.6 Ton 2 Star Split AC  - W...         2.0   
3     Voltas  Voltas 2023 Model 1.5 Ton 5 Star Split Inverte...         5.0   
4     Godrej  Godrej 5-In-1 Convertible Cooling 2023 Model 1...         3.0   
..       ...                                                ...         ...   
750       LG  LG 1.5 Ton 5 Star Split Dual Inverter AC  - White         5.0   
751   Daikin  Daikin 1.5 Ton 3 Star Split with PM 2.5 Filter...         3.0   
752  Hitachi    Hitachi 1 Ton 3 Star Split Inverter AC  - White         3.0   
753     Blue  Blue Star Convertible 5 in 1 Cooling 2023 Mode...         3.0   
754     Blue  Blue Star 2023 Model 1.3 Ton 3 Star Split Inve...         3.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
0     1.5           939         7,988          4.3        ₹43,900   
1     1.5           403         4,378          4.1        ₹49,990   
2     0.6            13           133          4.0        ₹49,690   
3     1.5           244         2,635          4.1        ₹75,990   
4     1.5           403         4,378          4.1        ₹45,400   
..    ...           ...           ...          ...            ...   
750   1.5             1            18          4.2        ₹83,990   
751   1.5           204         2,045          4.3        ₹43,800   
752   1.0             4            23          3.6            NaN   
753   1.5           384         4,478          4.1        ₹62,250   
754   1.3         1,521        13,889          4.2        ₹57,500   

    Current_Price  Room_Size  Annual_Power_Usage  
0         ₹29,990  111 - 150              969.15  
1         ₹32,990  111 - 150              883.14  
2         ₹19,999         90              475.30  
3         ₹37,990  111 - 150              751.28  
4         ₹30,990  111 - 150              965.71  
..            ...        ...                 ...  
750       ₹51,890  111 - 150              968.61  
751       ₹40,288  111 - 150             1425.00  
752       ₹41,290         90                0.00  
753       ₹35,799  111 - 150             1002.84  
754       ₹32,990  111 - 150              865.48  

[755 rows x 11 columns]

In [7]:
df.isna().sum()

Brands                  0
Model                   0
BEE_Rating              3
Tons                    1
Reviews_Count           0
Ratings_Count           0
Cust_Review             0
Original_Price        100
Current_Price           0
Room_Size               4
Annual_Power_Usage     19
dtype: int64

## Filling the missing values in Original Price column as while web scrapping there were some missing values for few products with no offers on them. So the original price of the product is the same as the current price

In [8]:
df['Original_Price'].fillna("x",inplace=True)

In [9]:
for i in range(df['Original_Price'].size):
     if df['Original_Price'][i]=="x":
            df['Original_Price'][i]=df['Current_Price'][i]

In [10]:
df['Original_Price'] = df['Original_Price']

In [11]:
df

Brands                                              Model  BEE_Rating  \
0      ONIDA  ONIDA 2023 Model 1.5 Ton 3 Star Split Inverter...         3.0   
1     Godrej  Godrej 5-in-1 Convertible Cooling 2023 Model 1...         4.0   
2     Voltas  Voltas 2023 Model 0.6 Ton 2 Star Split AC  - W...         2.0   
3     Voltas  Voltas 2023 Model 1.5 Ton 5 Star Split Inverte...         5.0   
4     Godrej  Godrej 5-In-1 Convertible Cooling 2023 Model 1...         3.0   
..       ...                                                ...         ...   
750       LG  LG 1.5 Ton 5 Star Split Dual Inverter AC  - White         5.0   
751   Daikin  Daikin 1.5 Ton 3 Star Split with PM 2.5 Filter...         3.0   
752  Hitachi    Hitachi 1 Ton 3 Star Split Inverter AC  - White         3.0   
753     Blue  Blue Star Convertible 5 in 1 Cooling 2023 Mode...         3.0   
754     Blue  Blue Star 2023 Model 1.3 Ton 3 Star Split Inve...         3.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
0     1.5           939         7,988          4.3        ₹43,900   
1     1.5           403         4,378          4.1        ₹49,990   
2     0.6            13           133          4.0        ₹49,690   
3     1.5           244         2,635          4.1        ₹75,990   
4     1.5           403         4,378          4.1        ₹45,400   
..    ...           ...           ...          ...            ...   
750   1.5             1            18          4.2        ₹83,990   
751   1.5           204         2,045          4.3        ₹43,800   
752   1.0             4            23          3.6        ₹41,290   
753   1.5           384         4,478          4.1        ₹62,250   
754   1.3         1,521        13,889          4.2        ₹57,500   

    Current_Price  Room_Size  Annual_Power_Usage  
0         ₹29,990  111 - 150              969.15  
1         ₹32,990  111 - 150              883.14  
2         ₹19,999         90              475.30  
3         ₹37,990  111 - 150              751.28  
4         ₹30,990  111 - 150              965.71  
..            ...        ...                 ...  
750       ₹51,890  111 - 150              968.61  
751       ₹40,288  111 - 150             1425.00  
752       ₹41,290         90                0.00  
753       ₹35,799  111 - 150             1002.84  
754       ₹32,990  111 - 150              865.48  

[755 rows x 11 columns]

In [12]:
df.isnull().sum()

Brands                 0
Model                  0
BEE_Rating             3
Tons                   1
Reviews_Count          0
Ratings_Count          0
Cust_Review            0
Original_Price         0
Current_Price          0
Room_Size              4
Annual_Power_Usage    19
dtype: int64

## Filling the missing values in the Tons column with the mean()

In [13]:
df[df["Tons"].isna()]

Brands                                              Model  BEE_Rating  \
143  Hitachi  Hitachi 1.5 T 5 Star Split Inverter AC  - Whit...         5.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
143   NaN           217         2,406          4.0        ₹65,300   

    Current_Price  Room_Size  Annual_Power_Usage  
143       ₹41,999  111 - 150               780.6

In [14]:
table1 = pd.pivot_table(df,index="Brands",values="Tons",aggfunc="mean")
table1

Tons
Brands             
Blue       1.291279
CANDY      1.375000
CARRIER    1.444444
Daikin     1.415556
Godrej     1.325000
Haier      1.405263
Hisense    1.000000
Hitachi    1.436667
IFB        1.400000
LG         1.427586
Lloyd      1.385714
MOTOROLA   1.750000
MarQ       1.237500
Midea      1.400000
O          1.312500
ONIDA      1.457778
Panasonic  1.353846
SAMSUNG    1.410714
Sansui     1.500000
TOSHIBA    1.500000
Thomson    1.000000
Voltas     1.422449
Whirlpool  1.363636
realme     1.400000

In [15]:
res = table1.loc['Hitachi',"Tons"].round(1)
res

1.4

In [16]:
df[df["Tons"].isna()]

Brands                                              Model  BEE_Rating  \
143  Hitachi  Hitachi 1.5 T 5 Star Split Inverter AC  - Whit...         5.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
143   NaN           217         2,406          4.0        ₹65,300   

    Current_Price  Room_Size  Annual_Power_Usage  
143       ₹41,999  111 - 150               780.6

In [17]:
df["Tons"][143]

nan

In [18]:
df.at[143,"Tons"] = res

In [19]:
df["Tons"][143]

1.4

In [20]:
df.isna().sum()

Brands                 0
Model                  0
BEE_Rating             3
Tons                   0
Reviews_Count          0
Ratings_Count          0
Cust_Review            0
Original_Price         0
Current_Price          0
Room_Size              4
Annual_Power_Usage    19
dtype: int64

------

## Filling the missing values in BEE_Ratings column with the mean()

In [22]:
test = df[df["BEE_Rating"].isna()]
test

Brands                                              Model  BEE_Rating  \
170    Blue  Blue Star 2023 Model 1 Ton Portable Turbo AC  ...         NaN   
290  Godrej            Godrej 2 Ton Split Inverter AC  - White         NaN   
333  Voltas            Voltas 2 Ton Split Inverter AC  - White         NaN   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
170   1.0             1             4          2.8        ₹39,000   
290   2.0             0             5          5.0        ₹60,999   
333   2.0             0             4          4.3        ₹79,990   

    Current_Price  Room_Size  Annual_Power_Usage  
170       ₹33,990         90                 NaN  
290       ₹40,999  151 - 200                 NaN  
333       ₹46,450  151 - 200                 NaN

In [23]:
null_indexs = test.index
null_indexs

Index([170, 290, 333], dtype='int64')

In [21]:
table2 = pd.pivot_table(df,index="Brands",columns="Tons",values="BEE_Rating",aggfunc="mean")
table2

Tons       0.60  0.70  0.75  0.80  0.90      1.00      1.20  1.30  1.40  \
Brands                                                                    
Blue        NaN   NaN   NaN   3.0   3.0  3.454545  3.000000   3.0   NaN   
CANDY       NaN   NaN   NaN   NaN   NaN  4.000000       NaN   NaN   NaN   
CARRIER     NaN   NaN   NaN   3.0   NaN  4.000000  5.000000   NaN   NaN   
Daikin      NaN   NaN   NaN   3.0   NaN  3.750000       NaN   NaN   NaN   
Godrej      NaN   NaN   NaN   NaN   NaN  3.945946       NaN   NaN   3.0   
Haier       NaN   NaN   NaN   NaN   NaN  2.750000  3.000000   3.0   NaN   
Hisense     NaN   NaN   NaN   NaN   NaN  5.000000       NaN   NaN   NaN   
Hitachi     NaN   NaN   NaN   NaN   NaN  3.285714       NaN   NaN   5.0   
IFB         NaN   NaN   NaN   NaN   NaN  3.000000       NaN   NaN   NaN   
LG          NaN   NaN   NaN   NaN   NaN  4.200000  3.000000   NaN   NaN   
Lloyd       NaN   NaN   NaN   3.0   NaN  3.727273  4.333333   NaN   NaN   
MOTOROLA    NaN   NaN   NaN   NaN   NaN       NaN       NaN   NaN   NaN   
MarQ        NaN   3.0   NaN   3.0   NaN  3.000000       NaN   NaN   NaN   
Midea       NaN   NaN   NaN   3.0   NaN       NaN       NaN   NaN   NaN   
O           NaN   NaN   4.0   NaN   NaN       NaN       NaN   NaN   NaN   
ONIDA       NaN   NaN   NaN   NaN   NaN  3.500000       NaN   NaN   NaN   
Panasonic   NaN   NaN   NaN   NaN   NaN  4.111111       NaN   NaN   NaN   
SAMSUNG     NaN   NaN   NaN   NaN   NaN  3.285714       NaN   NaN   NaN   
Sansui      NaN   NaN   NaN   NaN   NaN       NaN       NaN   NaN   NaN   
TOSHIBA     NaN   NaN   NaN   NaN   NaN       NaN       NaN   NaN   NaN   
Thomson     NaN   NaN   NaN   NaN   NaN  2.000000       NaN   NaN   NaN   
Voltas      2.0   NaN   3.0   NaN   NaN  3.444444  3.666667   NaN   NaN   
Whirlpool   NaN   NaN   NaN   NaN   NaN  4.000000       NaN   NaN   NaN   
realme      NaN   NaN   NaN   NaN   NaN       NaN  4.000000   NaN   NaN   

Tons           1.50  1.60  1.80      2.00      2.02  2.20  
Brands                                                     
Blue       4.056180   NaN   NaN  3.000000       NaN   NaN  
CANDY      4.666667   NaN   NaN       NaN       NaN   NaN  
CARRIER    4.333333   NaN   NaN  3.000000       NaN   NaN  
Daikin     3.411765   NaN   3.2       NaN  3.333333   NaN  
Godrej     3.904000   NaN   NaN  3.000000       NaN   NaN  
Haier      3.571429   4.5   NaN  3.000000       NaN   NaN  
Hisense         NaN   NaN   NaN       NaN       NaN   NaN  
Hitachi    4.000000   NaN   3.5  3.000000       NaN   NaN  
IFB        3.833333   3.0   NaN       NaN       NaN   NaN  
LG         4.500000   NaN   NaN  3.500000       NaN   NaN  
Lloyd      3.450000   NaN   NaN  3.666667       NaN   NaN  
MOTOROLA   3.000000   NaN   NaN  3.000000       NaN   NaN  
MarQ       3.666667   NaN   NaN       NaN       NaN   NaN  
Midea           NaN   NaN   NaN  3.000000       NaN   NaN  
O          4.333333   NaN   NaN       NaN       NaN   NaN  
ONIDA      3.100000   4.0   NaN       NaN       NaN   NaN  
Panasonic  4.000000   NaN   NaN       NaN       NaN   4.0  
SAMSUNG    3.894737   NaN   NaN  3.500000       NaN   NaN  
Sansui     5.000000   NaN   NaN       NaN       NaN   NaN  
TOSHIBA    5.000000   NaN   NaN       NaN       NaN   NaN  
Thomson         NaN   NaN   NaN       NaN       NaN   NaN  
Voltas     4.000000   NaN   NaN  3.714286       NaN   NaN  
Whirlpool  3.500000   NaN   NaN  3.000000       NaN   NaN  
realme     4.500000   NaN   NaN       NaN       NaN   NaN

In [24]:
table2[1.0]["Blue"]

3.4545454545454546

In [25]:
x = []
for i in test["Brands"]:
    x.append(i)
x

['Blue', 'Godrej', 'Voltas']

In [26]:
y = []
for k in test["Tons"]:
    y.append(k)
y

[1.0, 2.0, 2.0]

In [27]:
rats = []
for a,b in zip(x,y):
    rats.append(table2[b][a].round(0))
rats

[3.0, 3.0, 4.0]

In [31]:
df["BEE_Rating"][170]

nan

In [32]:
df["BEE_Rating"].value_counts()

BEE_Rating
3.0    425
5.0    243
4.0     70
2.0     14
Name: count, dtype: int64

In [33]:
df["BEE_Rating"][null_indexs]

170   NaN
290   NaN
333   NaN
Name: BEE_Rating, dtype: float64

In [34]:
for i in range(len(null_indexs)):
    df.at[null_indexs[i],"BEE_Rating"]=rats[i]

In [35]:
df["BEE_Rating"][null_indexs]

170    3.0
290    3.0
333    4.0
Name: BEE_Rating, dtype: float64

In [36]:
df.isna().sum()

Brands                 0
Model                  0
BEE_Rating             0
Tons                   0
Reviews_Count          0
Ratings_Count          0
Cust_Review            0
Original_Price         0
Current_Price          0
Room_Size              4
Annual_Power_Usage    19
dtype: int64

-----

## Filling the missing values in Room_Size with the mode()

In [37]:
table3 = df[df["Room_Size"].isna()]
table3

Brands                                              Model  BEE_Rating  \
244     Daikin  Daikin 2.02 Ton 3 Star Split Inverter AC with ...         3.0   
377     Daikin  Daikin 2.02 Ton 3 Star Split Inverter AC with ...         3.0   
413  Panasonic  Panasonic Convertible 7-in-1 with Additional A...         4.0   
435     Daikin  Daikin 2.02 Ton 4 Star Split Inverter AC with ...         4.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
244  2.02             0            11          4.7        ₹86,800   
377  2.02             0            11          4.7        ₹86,800   
413  2.20            78           638          4.1        ₹87,900   
435  2.02             0            18          3.4        ₹86,800   

    Current_Price Room_Size  Annual_Power_Usage  
244       ₹59,990       NaN             1372.85  
377       ₹59,990       NaN             1372.85  
413       ₹71,719       NaN             1351.04  
435       ₹62,990       NaN             1372.85

In [38]:
df[df["Brands"]=="Daikin"].groupby(by = "Tons")["Room_Size"].value_counts()

Tons  Room_Size
0.8   90            3
1.0   90            8
1.5   111 - 150    17
1.8   151 - 200     5
Name: count, dtype: int64

In [39]:
df[df["Brands"]=="Daikin"].groupby(by = "Tons")["Room_Size"].value_counts()[1.8]

Room_Size
151 - 200    5
Name: count, dtype: int64

In [40]:
df[df["Brands"]=="Panasonic"].groupby(by = "Tons")["Room_Size"].value_counts()

Tons  Room_Size
1.0   90            9
1.5   111 - 150    16
Name: count, dtype: int64

In [41]:
df[df["Brands"]=="Panasonic"].groupby(by = "Tons")["Room_Size"].value_counts()[1.5]

Room_Size
111 - 150    16
Name: count, dtype: int64

In [42]:
room_index = table3.index
room_index

Index([244, 377, 413, 435], dtype='int64')

In [43]:
table3

Brands                                              Model  BEE_Rating  \
244     Daikin  Daikin 2.02 Ton 3 Star Split Inverter AC with ...         3.0   
377     Daikin  Daikin 2.02 Ton 3 Star Split Inverter AC with ...         3.0   
413  Panasonic  Panasonic Convertible 7-in-1 with Additional A...         4.0   
435     Daikin  Daikin 2.02 Ton 4 Star Split Inverter AC with ...         4.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
244  2.02             0            11          4.7        ₹86,800   
377  2.02             0            11          4.7        ₹86,800   
413  2.20            78           638          4.1        ₹87,900   
435  2.02             0            18          3.4        ₹86,800   

    Current_Price Room_Size  Annual_Power_Usage  
244       ₹59,990       NaN             1372.85  
377       ₹59,990       NaN             1372.85  
413       ₹71,719       NaN             1351.04  
435       ₹62,990       NaN             1372.85

Replacing the missing values with

    Panasonic for 2.20 Tons = "111 - 150" (as the highest 1.5 tons)
    Dalkin for 2.0 Tons = "151 - 200" (as the highest 1.8 tons)

In [44]:
# df.at[413,"Room_Size"] = "111 - 150"

In [45]:
# df.at[435,"Room_Size"] = "151 - 200"
# df.at[377,"Room_Size"] = "151 - 200"
# df.at[244,"Room_Size"] = "151 - 200"

In [46]:
df["Room_Size"][room_index]

244    151 - 200
377    151 - 200
413    111 - 150
435    151 - 200
Name: Room_Size, dtype: object

In [47]:
df.isna().sum()

Brands                 0
Model                  0
BEE_Rating             0
Tons                   0
Reviews_Count          0
Ratings_Count          0
Cust_Review            0
Original_Price         0
Current_Price          0
Room_Size              0
Annual_Power_Usage    19
dtype: int64

-----------

## Filling the missing values in Annual_Power_Usage column with the mean()

In [48]:
table4 = df[df['Annual_Power_Usage'].isna()]
table4

Brands                                              Model  BEE_Rating  \
8         MarQ  MarQ by Flipkart 2024 Range 1 Ton 3 Star Split...         3.0   
38          LG       LG 1.5 Ton 5 Star Split Inverter AC  - White         5.0   
47        Blue  Blue Star 1.5 Ton 4 Star Split Inverter AC  - ...         4.0   
71        Blue  Blue Star 1.5 Ton 4 Star Split Inverter AC  - ...         4.0   
103         LG         LG 1 Ton 5 Star Split Inverter AC  - White         5.0   
138         LG         LG 1 Ton 3 Star Split Inverter AC  - White         3.0   
170       Blue  Blue Star 2023 Model 1 Ton Portable Turbo AC  ...         3.0   
187       Blue  Blue Star 1.5 Ton 4 Star Split Inverter AC  - ...         4.0   
290     Godrej            Godrej 2 Ton Split Inverter AC  - White         3.0   
333     Voltas            Voltas 2 Ton Split Inverter AC  - White         4.0   
460       Blue  Blue Star 1 Ton 5 Star Split Inverter AC  - White         5.0   
478     Voltas     Voltas 2 Ton 3 Star Split Inverter AC  - White         3.0   
485     Voltas     Voltas 2 Ton 5 Star Split Inverter AC  - White         5.0   
488      Lloyd            Lloyd 1 Ton 3 Star Portable AC  - White         3.0   
526       Blue  Blue Star 1.5 Ton 3 Star Split Inverter AC  - ...         3.0   
561     Daikin  Daikin 1 Ton 3 Star Split AC with PM 2.5 Filte...         3.0   
585     Daikin  Daikin 1.5 Ton 3 Star Split AC with PM 2.5 Fil...         3.0   
724  Panasonic  Panasonic 1.5 Ton 4 Star Split AC with Wi-fi C...         4.0   
738  Panasonic           Panasonic 1 Ton 3 Star Split AC  - White         3.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
8     1.0           104           981          4.3        ₹47,999   
38    1.5         2,889        28,367          4.2        ₹85,990   
47    1.5         1,521        13,889          4.2        ₹69,500   
71    1.5         1,521        13,889          4.2        ₹69,500   
103   1.0         2,889        28,367          4.2        ₹75,990   
138   1.0         2,889        28,367          4.2        ₹67,990   
170   1.0             1             4          2.8        ₹39,000   
187   1.5         1,521        13,889          4.2        ₹69,500   
290   2.0             0             5          5.0        ₹60,999   
333   2.0             0             4          4.3        ₹79,990   
460   1.0             0            10          4.7        ₹59,990   
478   2.0             5            39          4.0        ₹74,264   
485   2.0             1            50          4.3        ₹85,990   
488   1.0             4            27          3.3        ₹46,900   
526   1.5             1             4          4.0        ₹58,000   
561   1.0            20           159          4.2        ₹39,900   
585   1.5            48           532          4.3        ₹43,288   
724   1.5             0             5          3.6        ₹55,345   
738   1.0             7            34          4.4        ₹31,000   

    Current_Price  Room_Size  Annual_Power_Usage  
8         ₹26,490         90                 NaN  
38        ₹46,590  111 - 150                 NaN  
47        ₹40,990  111 - 150                 NaN  
71        ₹40,990  111 - 150                 NaN  
103       ₹39,990         90                 NaN  
138       ₹34,990         90                 NaN  
170       ₹33,990         90                 NaN  
187       ₹40,990  111 - 150                 NaN  
290       ₹40,999  151 - 200                 NaN  
333       ₹46,450  151 - 200                 NaN  
460       ₹39,990         90                 NaN  
478       ₹45,900  151 - 200                 NaN  
485       ₹55,800  151 - 200                 NaN  
488       ₹35,605         90                 NaN  
526       ₹42,500  111 - 150                 NaN  
561       ₹35,900         90                 NaN  
585       ₹42,500  111 - 150                 NaN  
724       ₹38,890  111 - 150                 NaN  
738       ₹31,000         90         

In [49]:
table5 = pd.pivot_table(df,index="Brands",columns=["Tons","BEE_Rating"],values="Annual_Power_Usage",aggfunc="mean")
table5

Tons         0.60    0.70    0.75                0.80    0.90        1.00  \
BEE_Rating    2.0     3.0     3.0     4.0         3.0     3.0         2.0   
Brands                                                                      
Blue          NaN     NaN     NaN     NaN  525.130556  644.46  765.333333   
CANDY         NaN     NaN     NaN     NaN         NaN     NaN         NaN   
CARRIER       NaN     NaN     NaN     NaN  526.030000     NaN         NaN   
Daikin        NaN     NaN     NaN     NaN  438.613333     NaN         NaN   
Godrej        NaN     NaN     NaN     NaN         NaN     NaN         NaN   
Haier         NaN     NaN     NaN     NaN         NaN     NaN  752.430000   
Hisense       NaN     NaN     NaN     NaN         NaN     NaN         NaN   
Hitachi       NaN     NaN     NaN     NaN         NaN     NaN         NaN   
IFB           NaN     NaN     NaN     NaN         NaN     NaN         NaN   
LG            NaN     NaN     NaN     NaN         NaN     NaN         NaN   
Lloyd         NaN     NaN     NaN     NaN  549.060000     NaN         NaN   
MOTOROLA      NaN     NaN     NaN     NaN         NaN     NaN         NaN   
MarQ          NaN  552.95     NaN     NaN  820.000000     NaN         NaN   
Midea         NaN     NaN     NaN     NaN  526.030000     NaN         NaN   
O             NaN     NaN     NaN  688.95         NaN     NaN         NaN   
ONIDA         NaN     NaN     NaN     NaN         NaN     NaN         NaN   
Panasonic     NaN     NaN     NaN     NaN         NaN     NaN         NaN   
SAMSUNG       NaN     NaN     NaN     NaN         NaN     NaN         NaN   
Sansui        NaN     NaN     NaN     NaN         NaN     NaN         NaN   
TOSHIBA       NaN     NaN     NaN     NaN         NaN     NaN         NaN   
Thomson       NaN     NaN     NaN     NaN         NaN     NaN  732.300000   
Voltas      475.3     NaN  727.66     NaN         NaN     NaN         NaN   
Whirlpool     NaN     NaN     NaN     NaN         NaN     NaN         NaN   
realme        NaN     NaN     NaN     NaN         NaN     NaN         NaN   

Tons                                          ...         1.80           \
BEE_Rating          3.0      4.0         5.0  ...          3.0      4.0   
Brands                                        ...                         
Blue         744.163333  574.355  550.090000  ...          NaN      NaN   
CANDY               NaN  600.210         NaN  ...          NaN      NaN   
CARRIER      704.460000      NaN  594.170000  ...          NaN      NaN   
Daikin       782.277500      NaN  488.533333  ...   352.403333      NaN   
Godrej       700.759231      NaN  524.030000  ...          NaN      NaN   
Haier        694.490000  609.480         NaN  ...          NaN      NaN   
Hisense             NaN      NaN  925.000000  ...          NaN      NaN   
Hitachi      587.990000      NaN  523.150000  ...  1251.000000  1096.51   
IFB          891.565000      NaN         NaN  ...          NaN      NaN   
LG           685.490000      NaN  622.225000  ...          NaN      NaN   
Lloyd        714.436667      NaN  554.980000  ...          NaN      NaN   
MOTOROLA            NaN      NaN         NaN  ...          NaN      NaN   
MarQ         736.070000      NaN         NaN  ...          NaN      NaN   
Midea               NaN      NaN         NaN  ...          NaN      NaN   
O                   NaN      NaN         NaN  ...          NaN      NaN   
ONIDA        702.306667      NaN  578.860000  ...          NaN      NaN   
Panasonic    686.480000  300.750  818.817500  ...          NaN      NaN   
SAMSUNG      695.365000      NaN    0.000000  ...          NaN      NaN   
Sansui              NaN      NaN         NaN  ...          NaN      NaN   
TOSHIBA             NaN      NaN         NaN  ...          NaN      NaN   
Thomson             NaN      NaN         NaN  ...          NaN      NaN   
Voltas      1224.398571      NaN  271.175000  ...          NaN      NaN   
Whirlpool    835.245000      NaN  515.270000 

In [134]:
table5[1.5][4.0]["Blue"]

1004.825

In [50]:
b_x = []
for i in table4["Brands"]:
    b_x.append(i)
b_x

['MarQ',
 'LG',
 'Blue',
 'Blue',
 'LG',
 'LG',
 'Blue',
 'Blue',
 'Godrej',
 'Voltas',
 'Blue',
 'Voltas',
 'Voltas',
 'Lloyd',
 'Blue',
 'Daikin',
 'Daikin',
 'Panasonic',
 'Panasonic']

In [51]:
len(b_x)

19

In [52]:
r_x = []
for j in table4["BEE_Rating"]:
    r_x.append(j)
r_x

[3.0,
 5.0,
 4.0,
 4.0,
 5.0,
 3.0,
 3.0,
 4.0,
 3.0,
 4.0,
 5.0,
 3.0,
 5.0,
 3.0,
 3.0,
 3.0,
 3.0,
 4.0,
 3.0]

In [53]:
t_x = []
for k in table4["Tons"]:
    t_x.append(k)
t_x

[1.0,
 1.5,
 1.5,
 1.5,
 1.0,
 1.0,
 1.0,
 1.5,
 2.0,
 2.0,
 1.0,
 2.0,
 2.0,
 1.0,
 1.5,
 1.0,
 1.5,
 1.5,
 1.0]

In [54]:
ann = []
for m,n,o in zip(t_x,r_x,b_x):
    ann.append(table5[m][n][o].round(2))
ann

[736.07,
 788.2,
 2070.0,
 2070.0,
 622.22,
 685.49,
 744.16,
 2070.0,
 1208.31,
 nan,
 550.09,
 1272.16,
 925.76,
 714.44,
 1030.96,
 782.28,
 952.97,
 876.76,
 686.48]

In [55]:
t4_ind = table4.index
t4_ind

Index([  8,  38,  47,  71, 103, 138, 170, 187, 290, 333, 460, 478, 485, 488,
       526, 561, 585, 724, 738],
      dtype='int64')

In [56]:
table4

Brands                                              Model  BEE_Rating  \
8         MarQ  MarQ by Flipkart 2024 Range 1 Ton 3 Star Split...         3.0   
38          LG       LG 1.5 Ton 5 Star Split Inverter AC  - White         5.0   
47        Blue  Blue Star 1.5 Ton 4 Star Split Inverter AC  - ...         4.0   
71        Blue  Blue Star 1.5 Ton 4 Star Split Inverter AC  - ...         4.0   
103         LG         LG 1 Ton 5 Star Split Inverter AC  - White         5.0   
138         LG         LG 1 Ton 3 Star Split Inverter AC  - White         3.0   
170       Blue  Blue Star 2023 Model 1 Ton Portable Turbo AC  ...         3.0   
187       Blue  Blue Star 1.5 Ton 4 Star Split Inverter AC  - ...         4.0   
290     Godrej            Godrej 2 Ton Split Inverter AC  - White         3.0   
333     Voltas            Voltas 2 Ton Split Inverter AC  - White         4.0   
460       Blue  Blue Star 1 Ton 5 Star Split Inverter AC  - White         5.0   
478     Voltas     Voltas 2 Ton 3 Star Split Inverter AC  - White         3.0   
485     Voltas     Voltas 2 Ton 5 Star Split Inverter AC  - White         5.0   
488      Lloyd            Lloyd 1 Ton 3 Star Portable AC  - White         3.0   
526       Blue  Blue Star 1.5 Ton 3 Star Split Inverter AC  - ...         3.0   
561     Daikin  Daikin 1 Ton 3 Star Split AC with PM 2.5 Filte...         3.0   
585     Daikin  Daikin 1.5 Ton 3 Star Split AC with PM 2.5 Fil...         3.0   
724  Panasonic  Panasonic 1.5 Ton 4 Star Split AC with Wi-fi C...         4.0   
738  Panasonic           Panasonic 1 Ton 3 Star Split AC  - White         3.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
8     1.0           104           981          4.3        ₹47,999   
38    1.5         2,889        28,367          4.2        ₹85,990   
47    1.5         1,521        13,889          4.2        ₹69,500   
71    1.5         1,521        13,889          4.2        ₹69,500   
103   1.0         2,889        28,367          4.2        ₹75,990   
138   1.0         2,889        28,367          4.2        ₹67,990   
170   1.0             1             4          2.8        ₹39,000   
187   1.5         1,521        13,889          4.2        ₹69,500   
290   2.0             0             5          5.0        ₹60,999   
333   2.0             0             4          4.3        ₹79,990   
460   1.0             0            10          4.7        ₹59,990   
478   2.0             5            39          4.0        ₹74,264   
485   2.0             1            50          4.3        ₹85,990   
488   1.0             4            27          3.3        ₹46,900   
526   1.5             1             4          4.0        ₹58,000   
561   1.0            20           159          4.2        ₹39,900   
585   1.5            48           532          4.3        ₹43,288   
724   1.5             0             5          3.6        ₹55,345   
738   1.0             7            34          4.4        ₹31,000   

    Current_Price  Room_Size  Annual_Power_Usage  
8         ₹26,490         90                 NaN  
38        ₹46,590  111 - 150                 NaN  
47        ₹40,990  111 - 150                 NaN  
71        ₹40,990  111 - 150                 NaN  
103       ₹39,990         90                 NaN  
138       ₹34,990         90                 NaN  
170       ₹33,990         90                 NaN  
187       ₹40,990  111 - 150                 NaN  
290       ₹40,999  151 - 200                 NaN  
333       ₹46,450  151 - 200                 NaN  
460       ₹39,990         90                 NaN  
478       ₹45,900  151 - 200                 NaN  
485       ₹55,800  151 - 200                 NaN  
488       ₹35,605         90                 NaN  
526       ₹42,500  111 - 150                 NaN  
561       ₹35,900         90                 NaN  
585       ₹42,500  111 - 150                 NaN  
724       ₹38,890  111 - 150                 NaN  
738       ₹31,000         90         

In [57]:
df["Annual_Power_Usage"][t4_ind]

8     NaN
38    NaN
47    NaN
71    NaN
103   NaN
138   NaN
170   NaN
187   NaN
290   NaN
333   NaN
460   NaN
478   NaN
485   NaN
488   NaN
526   NaN
561   NaN
585   NaN
724   NaN
738   NaN
Name: Annual_Power_Usage, dtype: float64

In [58]:
for i in range(len(t4_ind)):
    df.at[t4_ind[i],"Annual_Power_Usage"] = ann[i]

In [59]:
df["Annual_Power_Usage"][t4_ind]

8       736.07
38      788.20
47     2070.00
71     2070.00
103     622.22
138     685.49
170     744.16
187    2070.00
290    1208.31
333        NaN
460     550.09
478    1272.16
485     925.76
488     714.44
526    1030.96
561     782.28
585     952.97
724     876.76
738     686.48
Name: Annual_Power_Usage, dtype: float64

In [60]:
df.isna().sum()

Brands                0
Model                 0
BEE_Rating            0
Tons                  0
Reviews_Count         0
Ratings_Count         0
Cust_Review           0
Original_Price        0
Current_Price         0
Room_Size             0
Annual_Power_Usage    1
dtype: int64

In [62]:
table5[2.0][[3.0,4.0,5.0]]

BEE_Rating        3.0      4.0       5.0
Brands                                  
Blue        1384.0825      NaN       NaN
CANDY             NaN      NaN       NaN
CARRIER        0.0000      NaN       NaN
Daikin            NaN      NaN       NaN
Godrej      1208.3100      NaN       NaN
Haier       1276.5500      NaN       NaN
Hisense           NaN      NaN       NaN
Hitachi      698.8050      NaN       NaN
IFB               NaN      NaN       NaN
LG          1232.7800  1384.94       NaN
Lloyd       1154.3975      NaN  1020.225
MOTOROLA    1356.2300      NaN       NaN
MarQ              NaN      NaN       NaN
Midea       1249.8200      NaN       NaN
O                 NaN      NaN       NaN
ONIDA             NaN      NaN       NaN
Panasonic         NaN      NaN       NaN
SAMSUNG     1145.2200   880.00       NaN
Sansui            NaN      NaN       NaN
TOSHIBA           NaN      NaN       NaN
Thomson           NaN      NaN       NaN
Voltas      1272.1550      NaN   925.755
Whirlpool   1209.9800      NaN       NaN
realme            NaN      NaN       NaN

In [63]:
table6 = df[df["Annual_Power_Usage"].isna()]
table6

Brands                                    Model  BEE_Rating  Tons  \
333  Voltas  Voltas 2 Ton Split Inverter AC  - White         4.0   2.0   

    Reviews_Count Ratings_Count  Cust_Review Original_Price Current_Price  \
333             0             4          4.3        ₹79,990       ₹46,450   

     Room_Size  Annual_Power_Usage  
333  151 - 200                 NaN

In [64]:
df[(df["Brands"]=="Voltas")&(df["Tons"]==2.0)].groupby("BEE_Rating")["Annual_Power_Usage"].mean()

BEE_Rating
2.0    1542.790000
3.0    1272.156667
4.0            NaN
5.0     925.756667
Name: Annual_Power_Usage, dtype: float64

In [65]:
r3 = df[(df["Brands"]=="Voltas")&(df["Tons"]==2.0)].groupby("BEE_Rating")["Annual_Power_Usage"].mean()[3.0]
r3

1272.1566666666668

In [66]:
r5 = df[(df["Brands"]=="Voltas")&(df["Tons"]==2.0)].groupby("BEE_Rating")["Annual_Power_Usage"].mean()[5.0]
r5

925.7566666666667

In [67]:
r4 = ((r3+r5)/2).round(2)
r4

1098.96

In [68]:
t6_ind = table6.index
t6_ind

Index([333], dtype='int64')

In [69]:
df["Annual_Power_Usage"][t6_ind]

333   NaN
Name: Annual_Power_Usage, dtype: float64

In [70]:
for i in range(len(t6_ind)):
    df.at[t6_ind[i],"Annual_Power_Usage"]=r4

In [71]:
df["Annual_Power_Usage"][t6_ind]

333    1098.96
Name: Annual_Power_Usage, dtype: float64

In [72]:
df.isna().sum()

Brands                0
Model                 0
BEE_Rating            0
Tons                  0
Reviews_Count         0
Ratings_Count         0
Cust_Review           0
Original_Price        0
Current_Price         0
Room_Size             0
Annual_Power_Usage    0
dtype: int64

In [73]:
df

Brands                                              Model  BEE_Rating  \
0      ONIDA  ONIDA 2023 Model 1.5 Ton 3 Star Split Inverter...         3.0   
1     Godrej  Godrej 5-in-1 Convertible Cooling 2023 Model 1...         4.0   
2     Voltas  Voltas 2023 Model 0.6 Ton 2 Star Split AC  - W...         2.0   
3     Voltas  Voltas 2023 Model 1.5 Ton 5 Star Split Inverte...         5.0   
4     Godrej  Godrej 5-In-1 Convertible Cooling 2023 Model 1...         3.0   
..       ...                                                ...         ...   
750       LG  LG 1.5 Ton 5 Star Split Dual Inverter AC  - White         5.0   
751   Daikin  Daikin 1.5 Ton 3 Star Split with PM 2.5 Filter...         3.0   
752  Hitachi    Hitachi 1 Ton 3 Star Split Inverter AC  - White         3.0   
753     Blue  Blue Star Convertible 5 in 1 Cooling 2023 Mode...         3.0   
754     Blue  Blue Star 2023 Model 1.3 Ton 3 Star Split Inve...         3.0   

     Tons Reviews_Count Ratings_Count  Cust_Review Original_Price  \
0     1.5           939         7,988          4.3        ₹43,900   
1     1.5           403         4,378          4.1        ₹49,990   
2     0.6            13           133          4.0        ₹49,690   
3     1.5           244         2,635          4.1        ₹75,990   
4     1.5           403         4,378          4.1        ₹45,400   
..    ...           ...           ...          ...            ...   
750   1.5             1            18          4.2        ₹83,990   
751   1.5           204         2,045          4.3        ₹43,800   
752   1.0             4            23          3.6        ₹41,290   
753   1.5           384         4,478          4.1        ₹62,250   
754   1.3         1,521        13,889          4.2        ₹57,500   

    Current_Price  Room_Size  Annual_Power_Usage  
0         ₹29,990  111 - 150              969.15  
1         ₹32,990  111 - 150              883.14  
2         ₹19,999         90              475.30  
3         ₹37,990  111 - 150              751.28  
4         ₹30,990  111 - 150              965.71  
..            ...        ...                 ...  
750       ₹51,890  111 - 150              968.61  
751       ₹40,288  111 - 150             1425.00  
752       ₹41,290         90                0.00  
753       ₹35,799  111 - 150             1002.84  
754       ₹32,990  111 - 150              865.48  

[755 rows x 11 columns]

-----------

## Data Cleaning in each column and type conversions

In [74]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 755 entries, 0 to 754
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Brands              755 non-null    object 
 1   Model               755 non-null    object 
 2   BEE_Rating          755 non-null    float64
 3   Tons                755 non-null    float64
 4   Reviews_Count       755 non-null    object 
 5   Ratings_Count       755 non-null    object 
 6   Cust_Review         755 non-null    float64
 7   Original_Price      755 non-null    object 
 8   Current_Price       755 non-null    object 
 9   Room_Size           755 non-null    object 
 10  Annual_Power_Usage  755 non-null    float64
dtypes: float64(4), object(7)
memory usage: 65.0+ KB


In [75]:
df['Reviews_Count'].str.replace(",","")

0       939
1       403
2        13
3       244
4       403
       ... 
750       1
751     204
752       4
753     384
754    1521
Name: Reviews_Count, Length: 755, dtype: object

In [76]:
df["Reviews_Count"] = df['Reviews_Count'].str.replace(",","")

In [77]:
df["Reviews_Count"]= df["Reviews_Count"].astype("int")

In [78]:
df["Reviews_Count"]

0       939
1       403
2        13
3       244
4       403
       ... 
750       1
751     204
752       4
753     384
754    1521
Name: Reviews_Count, Length: 755, dtype: int32

----------

In [79]:
df['Ratings_Count']

0       7,988
1       4,378
2         133
3       2,635
4       4,378
        ...  
750        18
751     2,045
752        23
753     4,478
754    13,889
Name: Ratings_Count, Length: 755, dtype: object

In [80]:
df['Ratings_Count'] = df['Ratings_Count'].str.replace(",","")

In [81]:
df['Ratings_Count'] = df["Ratings_Count"].astype("int")

In [82]:
df['Ratings_Count']

0       7988
1       4378
2        133
3       2635
4       4378
       ...  
750       18
751     2045
752       23
753     4478
754    13889
Name: Ratings_Count, Length: 755, dtype: int32

In [83]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 755 entries, 0 to 754
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Brands              755 non-null    object 
 1   Model               755 non-null    object 
 2   BEE_Rating          755 non-null    float64
 3   Tons                755 non-null    float64
 4   Reviews_Count       755 non-null    int32  
 5   Ratings_Count       755 non-null    int32  
 6   Cust_Review         755 non-null    float64
 7   Original_Price      755 non-null    object 
 8   Current_Price       755 non-null    object 
 9   Room_Size           755 non-null    object 
 10  Annual_Power_Usage  755 non-null    float64
dtypes: float64(4), int32(2), object(5)
memory usage: 59.1+ KB


---------

In [84]:
df['Original_Price']

0      ₹43,900
1      ₹49,990
2      ₹49,690
3      ₹75,990
4      ₹45,400
        ...   
750    ₹83,990
751    ₹43,800
752    ₹41,290
753    ₹62,250
754    ₹57,500
Name: Original_Price, Length: 755, dtype: object

In [85]:
df['Original_Price'].str.replace("₹","")

0      43,900
1      49,990
2      49,690
3      75,990
4      45,400
        ...  
750    83,990
751    43,800
752    41,290
753    62,250
754    57,500
Name: Original_Price, Length: 755, dtype: object

In [86]:
df["Original_Price"] = df["Original_Price"].str.replace("₹","")

In [87]:
df['Original_Price'] = df["Original_Price"].str.replace(",","")

In [88]:
df["Original_Price"] = df['Original_Price'].astype("int")

In [89]:
df["Original_Price"]

0      43900
1      49990
2      49690
3      75990
4      45400
       ...  
750    83990
751    43800
752    41290
753    62250
754    57500
Name: Original_Price, Length: 755, dtype: int32

In [90]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 755 entries, 0 to 754
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Brands              755 non-null    object 
 1   Model               755 non-null    object 
 2   BEE_Rating          755 non-null    float64
 3   Tons                755 non-null    float64
 4   Reviews_Count       755 non-null    int32  
 5   Ratings_Count       755 non-null    int32  
 6   Cust_Review         755 non-null    float64
 7   Original_Price      755 non-null    int32  
 8   Current_Price       755 non-null    object 
 9   Room_Size           755 non-null    object 
 10  Annual_Power_Usage  755 non-null    float64
dtypes: float64(4), int32(3), object(4)
memory usage: 56.2+ KB


---------

In [91]:
df['Current_Price']

0      ₹29,990
1      ₹32,990
2      ₹19,999
3      ₹37,990
4      ₹30,990
        ...   
750    ₹51,890
751    ₹40,288
752    ₹41,290
753    ₹35,799
754    ₹32,990
Name: Current_Price, Length: 755, dtype: object

In [92]:
df['Current_Price'] = df['Current_Price'].str.replace("₹","")

In [93]:
df['Current_Price'] = df["Current_Price"].str.replace(",","")

In [94]:
df['Current_Price'] = df['Current_Price'].astype("int")

In [95]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 755 entries, 0 to 754
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Brands              755 non-null    object 
 1   Model               755 non-null    object 
 2   BEE_Rating          755 non-null    float64
 3   Tons                755 non-null    float64
 4   Reviews_Count       755 non-null    int32  
 5   Ratings_Count       755 non-null    int32  
 6   Cust_Review         755 non-null    float64
 7   Original_Price      755 non-null    int32  
 8   Current_Price       755 non-null    int32  
 9   Room_Size           755 non-null    object 
 10  Annual_Power_Usage  755 non-null    float64
dtypes: float64(4), int32(4), object(3)
memory usage: 53.2+ KB


In [96]:
df

Brands                                              Model  BEE_Rating  \
0      ONIDA  ONIDA 2023 Model 1.5 Ton 3 Star Split Inverter...         3.0   
1     Godrej  Godrej 5-in-1 Convertible Cooling 2023 Model 1...         4.0   
2     Voltas  Voltas 2023 Model 0.6 Ton 2 Star Split AC  - W...         2.0   
3     Voltas  Voltas 2023 Model 1.5 Ton 5 Star Split Inverte...         5.0   
4     Godrej  Godrej 5-In-1 Convertible Cooling 2023 Model 1...         3.0   
..       ...                                                ...         ...   
750       LG  LG 1.5 Ton 5 Star Split Dual Inverter AC  - White         5.0   
751   Daikin  Daikin 1.5 Ton 3 Star Split with PM 2.5 Filter...         3.0   
752  Hitachi    Hitachi 1 Ton 3 Star Split Inverter AC  - White         3.0   
753     Blue  Blue Star Convertible 5 in 1 Cooling 2023 Mode...         3.0   
754     Blue  Blue Star 2023 Model 1.3 Ton 3 Star Split Inve...         3.0   

     Tons  Reviews_Count  Ratings_Count  Cust_Review  Original_Price  \
0     1.5            939           7988          4.3           43900   
1     1.5            403           4378          4.1           49990   
2     0.6             13            133          4.0           49690   
3     1.5            244           2635          4.1           75990   
4     1.5            403           4378          4.1           45400   
..    ...            ...            ...          ...             ...   
750   1.5              1             18          4.2           83990   
751   1.5            204           2045          4.3           43800   
752   1.0              4             23          3.6           41290   
753   1.5            384           4478          4.1           62250   
754   1.3           1521          13889          4.2           57500   

     Current_Price  Room_Size  Annual_Power_Usage  
0            29990  111 - 150              969.15  
1            32990  111 - 150              883.14  
2            19999         90              475.30  
3            37990  111 - 150              751.28  
4            30990  111 - 150              965.71  
..             ...        ...                 ...  
750          51890  111 - 150              968.61  
751          40288  111 - 150             1425.00  
752          41290         90                0.00  
753          35799  111 - 150             1002.84  
754          32990  111 - 150              865.48  

[755 rows x 11 columns]

In [97]:
df.to_csv("Cleaned_AC_Data_Final.csv")